c/tskit/trees.c

```diff
@@ -2558,14 +2738,46 @@ sample_sets_to_bit_array(const tsk_treeseq_t *self, const tsk_size_t *sample_set
     return ret;
 }
 
+static int
+check_sites(const tsk_id_t *sites, tsk_size_t num_sites, tsk_size_t num_site_rows)
+{
+    int ret = 0;
+    tsk_id_t id;
+    tsk_size_t i;
+
+    if (sites == NULL) {
+        goto out;
+    }
+
+    for (i = 0; i < num_sites - 1; i++) {
+        id = sites[i];
+        if (id == TSK_NULL) {
+            ret = TSK_ERR_BAD_PARAM_VALUE; // TODO: new err type? TSK_BAD_SITE_VALUE?
+            goto out;
+        }
+        if (id > sites[i + 1]) {
```

Comment:
Does it matter if there are repeats? We're just specifying things to compute in the output matrix, right? I worry we're putting effort in to stop people doing "stupid" things, when this might also stop them doing clever things

Reply:
Repeats are going to complicate the indexing of the output array. We'd have to track the number of shared for rows and columns, we wouldn't be able to rely on their union. Also, we'd need a bit of rework when walking the trees because we assume that the sites are ascending and not repeating.

In short, I think it's possible to support repeats, but it's not currently supported. Do you think it's something we should be supporting?